lib/openhab/core_ext/ruby/time.rb

```diff
@@ -1,77 +1,106 @@
 # frozen_string_literal: true
 
-# Extensions to Time
-class Time
-  #
-  # @!method +(other)
-  #
-  # Extends {#+} to allow adding a {java.time.temporal.TemporalAmount TemporalAmount}
-  #
-  # @param [java.time.temporal.TemporalAmount] other
-  # @return [ZonedDateTime] If other is a {java.time.temporal.TemporalAmount TemporalAmount}
-  # @return [Time] If other is a Numeric
-  #
-  def plus_with_temporal(other)
-    return to_zoned_date_time + other if other.is_a?(java.time.temporal.TemporalAmount)
+require "forwardable"
 
-    plus_without_temporal(other)
-  end
-  alias_method :plus_without_temporal, :+
-  alias_method :+, :plus_with_temporal
+module OpenHAB
+  module CoreExt
+    module Ruby
+      # Extensions to Time and DateTime
+      module TimeExtensions
+        extend Forwardable
 
-  #
-  # @!method -(other)
-  #
-  # Extends {#-} to allow subtracting a {java.time.temporal.TemporalAmount TemporalAmount}
-  #
-  # @param [java.time.temporal.TemporalAmount] other
-  # @return [ZonedDateTime] If other is a {java.time.temporal.TemporalAmount TemporalAmount}
-  # @return [Time] If other is a Numeric
-  #
-  def minus_with_temporal(other)
-    return to_zoned_date_time - other if other.is_a?(java.time.temporal.TemporalAmount)
+        # @!visibility private
+        def self.included(base)
+          base.send :alias_method, :plus_without_temporal, :+
+          base.send :alias_method, :+, :plus_with_temporal
+          base.send :alias_method, :minus_without_temporal, :-
+          base.send :alias_method, :-, :minus_with_temporal
+        end
 
-    minus_without_temporal(other)
-  end
-  alias_method :minus_without_temporal, :-
-  alias_method :-, :minus_with_temporal
+        #
+        # @!method +(other)
+        #
+        # Extends {#+} to allow adding a {java.time.temporal.TemporalAmount TemporalAmount}
+        #
+        # @param [java.time.temporal.TemporalAmount] other
+        # @return [ZonedDateTime] If other is a {java.time.temporal.TemporalAmount TemporalAmount}
+        # @return [Time] If other is a Numeric
```

Comment:
What's the convention here to say return [base class] instead of Time?

Reply:
I haven't found one yet.